### PR TITLE
feat(secrets): add timeout for scan history checks

### PR DESCRIPTION
**Pull request by omryMen**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

## Description

adding a flag to the git history scan to stop scan.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/util/ext_argument_parser.py [hunk 1]
```diff
@@ -520,3 +520,9 @@ def add_parser_args(self) -> None:
             default=False,
             help="will scan the history of commits for secrets"
         )
+        self.add(
+            "--secrets-history-timeout",
+            action="append",
```
Comment (Eliran-Turgeman) [inline checkov/common/util/ext_argument_parser.py:525]: Can it be a list of values? or just a single value?

`action="append"` is useful for cases when you repeat the argument, if it is not needed you can remove this line as the default action is 'store'

https://docs.python.org/3/library/argparse.html#action

**Review by Eliran-Turgeman — APPROVED** [review]
🚀

File: checkov/main.py [hunk 2]
```diff
@@ -211,7 +211,8 @@ def run(banner: str = checkov_banner, argv: list[str] = sys.argv[1:]) -> int | N
                                  block_list_secret_scan=config.block_list_secret_scan,
                                  deep_analysis=config.deep_analysis,
                                  repo_root_for_plan_enrichment=config.repo_root_for_plan_enrichment,
-                                 enable_git_history_secret_scan=config.scan_secrets_history)
+                                 enable_git_history_secret_scan=config.scan_secrets_history,
+                                 git_history_timeout=config.secrets_history_timeout)
```
Comment (lirshindalman) [inline checkov/main.py:215]:
```suggestion
                                 enable_git_history_timeout=config.secrets_history_timeout)
```

**Review by lirshindalman — APPROVED** [review]
:ok_hand:

File: checkov/main.py (hunk 2 above)
Comment (omryMen) [reply checkov/main.py:215]: that's the actual timeout.. not a boolean. so enable doesn't make sense to me here

**Review by marynaKK — APPROVED** [review]
NICE! 🍒

**Pull request merged**